runtime/doc/lua.txt

```diff
@@ -130,7 +130,7 @@ Nvim tends to prefer the keyword args style.
 LUA PATTERNS                                                    *lua-patterns*
 
 Lua intentionally does not support regular expressions, instead it has limited
-"patterns" |luaref-patterns| which avoid the performance pitfalls of extended
+"patterns" |lua-patterns| which avoid the performance pitfalls of extended
```

Comment:
Doesn't this conflict with the tag on line 130?

`lua-patterns `is different to `luaref-patterns`. The reason `luaref` is used to to avoid any conflicts with tags in `lua.txt`.

Reply:
> Doesn't this conflict with the tag on line 130?

This link is wrong, it should be `lua-pattern` (singular).

> The reason `luaref` is used to to avoid any conflicts with tags in `lua.txt`.

I'm aware :) But it's not needed. 

- Almost all of the `luaref-x` tags cover topics that are owned by luaref.txt and don't have overlap with lua.txt. There is very little overlap.
- Anything that has overlap should be examined, typically by asking if a different tag name is better, or should it be moved, etc.
- lua-pattern/lua-patterns is the most salient case of "overlap". Doesn't justify the `luaref-` prefix.